presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonBasedUdfFunctionMetadata.java

```diff
@@ -50,27 +59,45 @@
      */
     private final RoutineCharacteristics routineCharacteristics;
 
+    /**
+     * Optional Aggregate-specific metadata (required for aggregation functions)
+     */
+    private final Optional<AggregationFunctionMetadata> aggregateMetadata;
+
     @JsonCreator
     public JsonBasedUdfFunctionMetadata(
             @JsonProperty("docString") String docString,
-            @JsonProperty("outputType") String outputType,
-            @JsonProperty("paramTypes") List<String> paramTypes,
+            @JsonProperty("functionKind") FunctionKind functionKind,
+            @JsonProperty("outputType") TypeSignature outputType,
+            @JsonProperty("paramTypes") List<TypeSignature> paramTypes,
             @JsonProperty("schema") String schema,
-            @JsonProperty("routineCharacteristics") RoutineCharacteristics routineCharacteristics)
+            @JsonProperty("routineCharacteristics") RoutineCharacteristics routineCharacteristics,
+            @JsonProperty("aggregateMetadata") Optional<AggregationFunctionMetadata> aggregateMetadata)
     {
         this.docString = requireNonNull(docString, "docString is null");
+        this.functionKind = requireNonNull(functionKind, "functionKind is null");
         this.outputType = requireNonNull(outputType, "outputType is null");
         this.paramTypes = ImmutableList.copyOf(requireNonNull(paramTypes, "paramTypes is null"));
         this.schema = requireNonNull(schema, "schema is null");
         this.routineCharacteristics = requireNonNull(routineCharacteristics, "routineCharacteristics is null");
+
+        if (functionKind == AGGREGATE && (aggregateMetadata == null || !aggregateMetadata.isPresent())) {
```

Comment:
argument should either be optional or nullable, but not both. The class field we can leave as an optional, and the function argument, it just depends what happens by default if you don't specify aggregateMetadata in the json file.

Reply:
This was added to guard against direct instantiation of this class using the ctor. Will add a `requireNonNull` condition & use a `checkArgument` ( w/  `aggregateMetadata == null` removed)